stmhal/dac.c

```diff
@@ -171,6 +171,14 @@ STATIC mp_obj_t pyb_dac_init_helper(pyb_dac_obj_t *self, mp_uint_t n_args, const
     #endif
 
     // stop anything already going on
+    __DMA1_CLK_ENABLE();
+    DMA_HandleTypeDef DMA_Handle;
```

Comment:
To be honest the fix was based more on guesswork than on a detailed understanding of the DMA system. I copied the code from lines 401-426 at the start of pyb_dac_write_timed() on the assumption that it was a proven way to shut down DMA. I was hoping for review/guidance from those with a better understanding of the mechanism.